spec/javascripts/Dashboard/ApiFilter.spec.jsx

```diff
@@ -22,12 +22,28 @@ it('should render itself', () => {
   expect(apiFilter.find('.ApiFilter').exists()).toBe(true)
 })
 
-it.skip('should filter APIs passed in props by name', () => {
-  // TODO
+it('should filter APIs passed in props by name', () => {
+  // remove / add mean class 'hidden' from DOM Element
+  const remove = jest.fn()
+  const add = jest.fn()
+  jest.spyOn(document, 'getElementById').mockReturnValue({
```

Comment:
This smells a bit. I know where it comes from in this case, there's invoked inside the `ApiFilter` component, inside a fn that receives an Api list, then for every item it looks for the element and removes the class according the filter.... I'd change the design of this making it less coupled: the "filter" could be stored in the parent component state and you render the children filtering the Api array, without the need of using classes.

Reply:
Problem is there is no parent component.. the list is [rendered by rails](https://github.com/3scale/porta/blob/60a39606a3dd26060ab1d1f1ba65aec919f4633b/app/views/provider/admin/dashboards/_section_services.html.slim#L5). This PR's scope is for the existing `ApiFilter`, since tests were [left out in the original PR](https://github.com/3scale/porta/pull/1058#discussion_r316262016). Maybe the question is where these tests should be implemented in gherkin/ruby instead of jest.

I'd leave it as smelly as it is because of time constraints and leave the proper implementation for porta fly.

Reply:
Fair enough :)